File: python/7 kyu/187.py
# https://www.codewars.com/kata/557592fcdfc2220bed000042/train/python

# Passed

def pattern(n: int) -> str:
    cycles = []
    numbers = [str(n) for n in range(1, n + 1)]
    for idx in range(n):
        idx %= len(numbers)
        cycle_pattern = "".join(numbers[idx:]) + "".join(numbers[:idx])
        cycles.append(cycle_pattern)
        
    result = "\n".join(cycles)
    return result

output = pattern(11)
print(output)

File: nodejs/7 kyu/187.js
// https://www.codewars.com/kata/525e5a1cb735154b320002c8/train/javascript

// Passed

function triangular(n) {
  if (n <= 0) {
    return 0;
  }

  let triangularNumber = (n * (n + 1)) / 2;
  return triangularNumber;
}

const output = triangular(4)
console.log(output)